From 337dafaa0d7bf49e25b4ec74ff44ca697abd06bc Mon Sep 17 00:00:00 2001
From: Ken Zhu <quic_guigenz@quicinc.com>
Date: Tue, 2 May 2023 21:39:54 -0700
Subject: [PATCH 213/281] bridge: fix bridge vlan filter toggle issue

when vlan filter is disabled, it could cause new bridge
creation failed if vlan filter setting existing in the
netlink creation message

Change-Id: I02f010e746e94623fda544846d05908a83d14fa2
Signed-off-by: Ken Zhu <quic_guigenz@quicinc.com>
---
 net/bridge/br_netlink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/bridge/br_netlink.c b/net/bridge/br_netlink.c
index 35b305921cdf..d838b7f631a4 100644
--- a/net/bridge/br_netlink.c
+++ b/net/bridge/br_netlink.c
@@ -1265,6 +1265,7 @@ static int br_changelink(struct net_device *brdev, struct nlattr *tb[],
 		br_stp_set_bridge_priority(br, priority);
 	}
 
+#ifdef CONFIG_BRIDGE_VLAN_FILTERING
 	if (data[IFLA_BR_VLAN_FILTERING]) {
 		u8 vlan_filter = nla_get_u8(data[IFLA_BR_VLAN_FILTERING]);
 
@@ -1273,7 +1274,6 @@ static int br_changelink(struct net_device *brdev, struct nlattr *tb[],
 			return err;
 	}
 
-#ifdef CONFIG_BRIDGE_VLAN_FILTERING
 	if (data[IFLA_BR_VLAN_PROTOCOL]) {
 		__be16 vlan_proto = nla_get_be16(data[IFLA_BR_VLAN_PROTOCOL]);
 
-- 
2.17.1

